aten/src/ATen/native/cuda/ScanUtils.cuh

```diff
@@ -343,12 +343,12 @@ __device__ void tensor_kernel_scan_innermost_dim_impl(T* row_buf, T *tgt_, const
 
       // Parallel reduction with Sklansky method. The diagram can be seen on this paper:
       // https://research.nvidia.com/publication/single-pass-parallel-prefix-scan-decoupled-look-back
-      for (uint32_t m = 0; m <= log_num_threads_x; ++m) {
+      for (int m = 0; m <= log_num_threads_x; ++m) {
         if (row_exists) {
-          uint32_t s = 1 << m; // s = 2 ^ m
-          uint32_t a = ((threadIdx.x >> m) << (m + 1)) | s; // a = (threadIdx.x / s) * (2 * s) + s
-          uint32_t ti = a + (threadIdx.x % s);
-          uint32_t si = a - 1;
+          index_t s = 1 << m; // s = 2 ^ m
```

Comment:
similarly here, all these indices are uint32_t, they index into shared memory array that can't be large